service/DeliveryServiceModel.go

```diff
@@ -57,3 +61,48 @@ func (s *DeliveryService) UpdateDeliveryService(service dao.DeliveryService) err
 	}
 	return nil
 }
+
+func (s *DeliveryService) SaveLogoFile(cover []byte, id int) error {
+	ACCESS_KEY := os.Getenv("ACCESS_KEY")
+	SECRET_KEY := os.Getenv("SECRET_KEY")
+	endpoint := "fra1.digitaloceanspaces.com"
+	ssl := true
+
+	// Initiate a client using DigitalOcean Spaces.
+	client, err := minio.New(endpoint, ACCESS_KEY, SECRET_KEY, ssl)
+	if err != nil {
+		log.Fatal(err)
+		return err
+	}
+
+	// List all Spaces.
+	spaces, err := client.ListBuckets()
+	if err != nil {
+		log.Println(err)
+		return err
+	}
+	for _, space := range spaces {
+		log.Println(space.Name)
+	}
+
+	_, err1 := client.PutObject("storage-like-s3", fmt.Sprintf("logo_img/%s", strconv.Itoa(id)),
+		bytes.NewReader(cover), int64(len(cover)), minio.PutObjectOptions{ContentType: "image/jpeg", UserMetadata: map[string]string{"x-amz-acl": "public-read"}})
+	if err1 != nil {
+		log.Println(err1)
+		return err1
+	}
+
+	var service dao.DeliveryService
+	service.Id = id
+	service.Photo = "https://storage-like-s3.fra1.digitaloceanspaces.com/logo_img/" + strconv.Itoa(id)
+
+	log.Println("https://storage-like-s3.fra1.digitaloceanspaces.com/logo_img/" + strconv.Itoa(id))
+
+	if err := s.repo.UpdateDeliveryServiceInDB(service); err != nil {
+		log.Println(err)
+		return fmt.Errorf("Error in DeliveryService: %s", err)
+	}
+
+	log.Println("Downloaded cover")
```

Comment:
You meant uploaded, not downloaded? Also it is better to add more info in the message, at least service id, but better image path or so on

Reply:
fixed, thanks!